examples/mupdf-react/src/viewer.js

```diff
@@ -0,0 +1,782 @@
+}
+
+function set_search_needle(needle) {
+	search_status.textContent = ""
+	current_search_needle = needle
+
+	if (!page_list)
+		return
+
+	for (let page of page_list)
+		page.setSearch(current_search_needle)
+
+	queue_update_view()
+}
+
+async function run_search(direction, step) {
+	// start search from visible page
+	set_search_needle(search_input.value)
+
+	current_search_page = find_visible_page()
+
+	let next_page = current_search_page
+	if (step)
+		next_page += direction
+
+	while (next_page >= 0 && next_page < page_list.length) {
+		// We run the check once per loop iteration,
+		// in case the search was cancel during the 'await' below.
+		if (current_search_needle === "") {
+			search_status.textContent = ""
+			return
+		}
+
+		search_status.textContent = `Searching page ${next_page}.`
+
+		if (page_list[next_page].loadNeedle !== page_list[next_page].needle)
+			await page_list[next_page]._loadSearch()
+
+		const hits = page_list[next_page].searchData
+		if (hits && hits.length > 0) {
+			page_list[next_page].rootNode.scrollIntoView()
+			current_search_page = next_page
+			search_status.textContent = `${hits.length} hits on page ${next_page}.`
+			return
+		}
+
+		next_page += direction
+	}
+
+	search_status.textContent = "No more search hits."
+}
+
+// OUTLINE
+
+function build_outline(parent, outline) {
+	for (let item of outline) {
+		let node = document.createElement("li")
+		let a = document.createElement("a")
+		a.href = "#page" + (item.page + 1)
+		a.textContent = item.title
+		node.appendChild(a)
+		if (item.down) {
+			let down = document.createElement("ul")
+			build_outline(down, item.down)
+			node.appendChild(down)
+		}
+		parent.appendChild(node)
+	}
+}
+
+function toggle_outline_panel() {
+	if (document.getElementById("outline-panel").style.display === "none")
+		show_outline_panel()
+	else
+		hide_outline_panel()
+}
+
+function show_outline_panel() {
+	if (!page_list)
+		return
+	document.getElementById("outline-panel").style.display = "block"
+}
+
+function hide_outline_panel() {
+	document.getElementById("outline-panel").style.display = "none"
+}
+
+// DOCUMENT LOADING
+
+function close_document() {
+	clear_message()
+	hide_outline_panel()
+	hide_search_panel()
+
+	if (current_doc) {
+		worker.closeDocument(current_doc)
+		current_doc = 0
+		document.getElementById("outline").replaceChildren()
+		document.getElementById("pages").replaceChildren()
+		for (let page of page_list)
+			page_observer.unobserve(page.rootNode)
+		page_visible.length = 0
+	}
+
+	page_list = null
+}
+
+async function open_document_from_buffer(buffer, magic, title) {
+	current_doc = await worker.openDocumentFromBuffer(buffer, magic)
+
+	document.title = await worker.documentTitle(current_doc) || title
+
+	page_count = await worker.countPages(current_doc)
+
+	// Use second page as default page size (the cover page is often differently sized)
+	var page_size = await worker.getPageSize(current_doc, page_count > 1 ? 2 : 1)
```

Comment:
Out of index by 1. Change to `1 ? 1 : 0` ( note this is also a bug in the current simple viewer, https://github.com/ArtifexSoftware/mupdf.js/commit/91228c47b452d27e9e106ff5353afb9e6fe26f63 )